core/src/main/scala/cats/instances/future.scala

```diff
@@ -8,12 +8,18 @@ import scala.concurrent.{ExecutionContext, Future}
 
 trait FutureInstances extends FutureInstances1 {
 
-  implicit def catsStdInstancesForFuture(implicit ec: ExecutionContext): MonadError[Future, Throwable] with CoflatMap[Future] =
-    new FutureCoflatMap with MonadError[Future, Throwable]{
+  implicit def catsStdInstancesForFuture(implicit ec: ExecutionContext): MonadError[Future, Throwable] with CoflatMap[Future] with MonadRec[Future] =
+    new FutureCoflatMap with MonadError[Future, Throwable] with MonadRec[Future] {
       def pure[A](x: A): Future[A] = Future.successful(x)
 
       def flatMap[A, B](fa: Future[A])(f: A => Future[B]): Future[B] = fa.flatMap(f)
 
+      final def tailRecM[B, C](b: B)(f: B => Future[(B Xor C)]): Future[C] =
```

Comment:
@kailuowang This implementation won't actually compile with `@tailrec`, since the recursive call is inside the `flatMap`. It is stack-safe, though, thanks to the way `Future#flatMap` works.